src/include/gurt/common.h

```diff
@@ -337,29 +337,25 @@ d_realpath(const char *path, char *resolved_path) _dalloc_;
 		d_errno2der(_rc);                                                                  \
 	})
 
-#define __D_PTHREAD_TIMED(fn, fn2, x)                                                              \
+#define __D_PTHREAD_TIMED(fn, x)                                                                   \
 	({                                                                                         \
-		int _rc;                                                                           \
-		_rc = fn((x));                                                                     \
-		if (_rc == EBUSY) {                                                                \
-			int             delay1 = 0;                                                \
-			int             delay2 = 1;                                                \
-			int             f      = delay1 + delay2;                                  \
-			int             c      = f;                                                \
-			struct timespec _wait  = {};                                               \
+		int             _rc;                                                               \
+		int             _delay1 = 0;                                                       \
+		int             _delay2 = 1;                                                       \
+		int             _f      = _delay1 + _delay2;                                       \
+		int             _c      = _f;                                                      \
+		struct timespec _wait   = {};                                                      \
+		clock_gettime(CLOCK_REALTIME, &_wait);                                             \
```

Comment:
Reading manpage on timed locks i see this:
"Under no  circumstances  shall  the  function  fail  with  a timeout if the lock can be
       acquired immediately. The validity of the abstime parameter need not be
       checked if the lock can be immediately acquired."

Perhaps then in order to avoid clock_gettime() on a happy path, the first time you can call fn() with null wait, and if it fails only then do clock_gettime() and proper time calc.

Having said that, i dont think doing gettime is expensive as it goes through some specially mmapped page instead of doing full system call iirc.

Reply:
I'm not worried about the cost of clock_gettime(), I just wrote this as I did and hadn't thought through how to simplfy it.  Thank you for the feedback.